### PR TITLE
Fixes #9270 - Add email validation

### PR DESCRIPTION
**Pull request by dLobatog**
Copied the email validation we use for the User model. This catches
errors when creating/updating the Docker compute resource email

### CODE REVIEW

**Comment by dLobatog** [comment]
@daviddavis @witlessbird @parthaa Can any of you take a quick look? This one is *easy* :smile:

File: app/models/foreman_docker/docker.rb [hunk 1]
```diff
@@ -5,6 +5,10 @@ class Docker < ::ComputeResource
     attr_accessible :email
 
     validates :url, :format => { :with => URI.regexp }
+    validates :email, :format => { :with => /\A(([\w!#\$%&\'\*\+\-\/=\?\^`\{\|\}~]+((\.\"[\w!#\$%&\'\*\+\-\/=\?\^`\{\|\}~\"\(\),:;<>@\[\\\] ]+(\.[\w!#\$%&\'\*\+\-\/=\?\^`\{\|\}~\"\(\),:;<>@\[\\\] ]+)*\")*\.[\w!#\$%&\'\*\+\-\/=\?\^`\{\|\}~]+)*)|(\"[\w !#\$%&\'\*\+\-\/=\?\^`\{\|\}~\"\(\),:;<>@\[\\\] ]+(\.[\w !#\$%&\'\*\+\-\/=\?\^`\{\|\}~\"\(\),:;<>@\[\\\] ]+)*\"))
```
Comment (stbenjam) [inline app/models/foreman_docker/docker.rb]: That regex tho

https://davidcel.is/posts/stop-validating-email-addresses-with-regex/

Comment (stbenjam) [inline app/models/foreman_docker/docker.rb]: It's just an opinion, of course, but one I agree with.  But if we're going to validate e-mail addresses then we might as well do it everywhere.  Can't we keep this monstrosity of a regex in one place? Like app/validators in core?

File: app/models/foreman_docker/docker.rb [hunk 2]
```diff
@@ -5,6 +5,10 @@ class Docker < ::ComputeResource
...
+                                  @[a-z0-9]+((\.[a-z0-9]+)*|(\-[a-z0-9]+)*)*\z/ix },
+    :length => { :maximum => 60 },
```
Comment (stbenjam) [inline app/models/foreman_docker/docker.rb]: Hmm, where does this 60 come from? Docker itself? Certainly no such limitation in the RFC's.

Comment (dLobatog) [inline app/models/foreman_docker/docker.rb]: It doesn't really come from anywhere, it's copied from user.rb https://github.com/theforeman/foreman/blob/develop/app/models/user.rb#L75

File: app/models/foreman_docker/docker.rb (hunk 1 above)
Comment (dLobatog) [inline app/models/foreman_docker/docker.rb]: Yup. I intended to do so in the first place, however by the time that's released so I can use it here, the BZ should be closed :disappointed:

File: app/models/foreman_docker/docker.rb (hunk 2 above)
Comment (stbenjam) [inline app/models/foreman_docker/docker.rb]: I filed http://projects.theforeman.org/issues/14641, and someone else had trouble with the 60-length limit before according to http://projects.theforeman.org/issues/9584.

If you want to keep the regex, that's fine, it does seem to validate everything including weird locals like `"let's send mail"@gmail.com.com`, but I'd at least remove the the length limitation, or increase it to be 254.

Comment (dLobatog) [inline app/models/foreman_docker/docker.rb]: Aight, I tend to agree with that reasoning. I'll shorten the regex & remove the chars limitation

**Comment by dLobatog** [comment]
Merged as 977f7c5a137d6e1210d6c3e77db6f0f6763feaa9, thanks @dLobatog!

**Pull request closed without merging**